src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str.cs

```diff
@@ -1,221 +1,219 @@
+            Assert.NotNull(GetEntries(Directory.GetCurrentDirectory()));
+        }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)]
-    public void WildCharactersFileName()
-    {
-        String strTempDir = Path.Combine("dls;d", "442349-0", "v443094(*)(+*$#$*") + new string(Path.DirectorySeparatorChar, 3);
-        Assert.Throws<ArgumentException>(() => GetEntries(strTempDir));
-    }
+        #endregion
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)]
-    public void WindowsFileNameWithSpaces()
-    {
-        Assert.Throws<ArgumentException>(() => GetEntries("\n"));
-        Assert.Throws<ArgumentException>(() => GetEntries("          "));
-        Assert.Throws<ArgumentException>(() => GetEntries(" "));
-        Assert.Throws<ArgumentException>(() => GetEntries(""));
-        Assert.Throws<ArgumentException>(() => GetEntries(">"));
-        Assert.Throws<ArgumentException>(() => GetEntries("\0"));
-    }
+        #region PlatformSpecific
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixFileNameWithSpaces()
-    {
-        String testDirPath = Path.Combine(TestDirectory, GetTestFileName());
-        Directory.CreateDirectory(testDirPath);
-        using (File.Create(Path.Combine(testDirPath, " ")))
-        using (File.Create(Path.Combine(testDirPath, "          ")))
-        using (File.Create(Path.Combine(testDirPath, "\n")))
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWildCharactersFileName()
         {
-            String[] results = GetEntries(testDirPath);
-            Assert.Contains(Path.Combine(testDirPath, " "), results);
-            Assert.Contains(Path.Combine(testDirPath, "          "), results);
-            Assert.Contains(Path.Combine(testDirPath, "\n"), results);
+            string strTempDir = Path.Combine("dls;d", "442349-0", "v443094(*)(+*$#$*") + new string(Path.DirectorySeparatorChar, 3);
+            Assert.Throws<ArgumentException>(() => GetEntries(strTempDir));
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixDirectoryNameWithSpaces()
-    {
-        String testDirPath = Path.Combine(TestDirectory, GetTestFileName());
-        DirectoryInfo testDir = new DirectoryInfo(testDirPath);
-        testDir.CreateSubdirectory(" ");
-        testDir.CreateSubdirectory("\n");
-        testDir.CreateSubdirectory("          ");
-
-        String[] results = GetEntries(testDirPath);
-        Assert.Contains(Path.Combine(testDirPath, " "), results);
-        Assert.Contains(Path.Combine(testDirPath, "          "), results);
-        Assert.Contains(Path.Combine(testDirPath, "\n"), results);
-
-        testDir.Delete(true);
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWhitespacePath()
+        {
+            foreach (string whitespace in WhitespaceVariations)
+                Assert.Throws<ArgumentException>(() => GetEntries(whitespace));
```

Comment:
This could be done using Assert.All, e.g.
```C#
Assert.All(WhitespaceVariations, whitespace =>
    Assert.Throws<ArgumentException>(() => GetEntries(whitespace)));
```